basics/5 terminal/index.md

```diff
@@ -1,27 +1,147 @@
+
+> ### Pro tip!
+> Don't feel like typing out a long path to a directory? Simply drag the folder onto your Terminal from your Finder or Explorer. The full path will apear right away!
 
 ## Creating a new repo on Github and cloning
+Now, let's get started using git. Git is a great tool for managing programming projects, from small projects to enterprise-level applications. It keeps track of all changes, and allows for you to work on a project with a team of people, all while avoiding conlicts as much as possible.
+
+Your code in git is placed in a so-called 'repository'. Within a repository, git keeps track of changes and resolved conflicts. Every repository needs to be hosted somehwere. Ofcourse, you could host it on your own device. If you're going to work with other people, however, this won't work. Most repository's, therefore, are primarily hosted through services like [Github](https://github.com). You should already have an account there. If not, create one now!
+
+Once you log in to Github, you end up in your dashboard. From here, you can create a new repository by hitting the 'New' button on the left of the screen.
+
+![](github_dash.png)
+
+Give your repository a name (it must be unique within your account), a description and select whether you want it to be public or private. Public repository's, including all the code in them, can be seen by anyone on the internet. Private repository's can only be seen by the people you add to them. Finally, tick the box next to 'Add a README file'.
+
+![](github_new.png)
+
+You now have a new Github repository. It's empty, apart from the README file. So let's make sure you can actually work on your code within the repository, on your own device. To do so, you must 'Clone' the repository by clicking on the green 'Code' button and copying the URL that appears. 
+
+![github_clone.png]()
+
+Then, in a Terminal window, navigate to a folder where you want your code to live. Then, run `git clone [URL]`, replacing `[URL]` with the URL you just copied. Git should perform some magic, and a new folder with the name of your repository should appear. In there, you'll find the README file from Github!
+
+![](git_clone.png)
 
 ## Updating, committing and pushing
+Now, we want to actually host some code on Github. In your Terminal, navigate to the new repository folder with the README in it and enter `touch hello.py`. A new file, named 'hello.py', should appear. Then. write some code in there. I'm just adding some code to ask someone their name, and tell them hello.
+
+![](hello.png)
+
+Now, we want to tell git that we actually want it to track this file. We can do so by running `git add hello.py` within our repository folder.
+
+> ### Adding a lot of files?
+> If you've just added or updated a lot of files, and want to add them all to git, just run `git add *`. All files will be added at once.
```

Comment:
`git add .` is betrouwbaarder, in die zin dat `git add *` eventuele "dotfiles" overslaat, waaronder `.gitignore`

Die `.` voor cwd moeten we dan ook even uitleggen (eerder al misschien en dan hier herhalen).